### ISSUE TITLE
Cleanup, Refactor, Optimize!

### ISSUE THREAD

**Issue by zMyst22** [zMyst22/WhiteBoxNeuralNetwork#7]
File is full of old and abandon code. I should also do something about setting parameters so I don't need to comment and uncomment to do things like creating new networks. As of now, the actual backpropagation steps are essentially hardcoded at the moment, restricting the network to having a single hidden layer (but still i'm so hyped i got it working ^-^). There are also some other optimizations needed before I can move onto alternative processing methods. 
Here's the plan:
### Cleanup
1. Lot of old code and abandoned functions to be removed
2. Remove unnecessary comments
3. Implicate more comments on the function of the code
### Refactor
1. Reorganize code for readability
2. Refactor the code to work recursively and automatically depending on the number of inputs, outputs, and layers (right now it only works with a single hidden layer)
3. Construct a modular system that I am able to effectively predict and optimize fluidly
### Optimize
1. I want to start simple with using multi-threading and multi-processing, but it's not exactly "real" with python but its a start.
2. Possibly utilize more RAM for the time being, as these loads shouldn't be more than a few gigs. But I still need to plan ahead for more complex data structures. I want everything to be able to move forward with additions, not having to rebuild everything when I want to make an upgrade. I need a simple and strong build that can equip any armor I throw on it.
3. Once the base system is complete, it'll open the door way for C and/or shader processing (yes I'm aware Python technically IS C, but to keep as much of it in real C will increase performance).
4. **The refactoring step is the preparation stage for optimization. This portion of optimizing is to unlock as much horsepower as possible before I implement any of the aforementioned upgrades so ensure complete efficiency in each step.**

**Comment by zMyst22** [comment]
Calling this one Case: CLOSED
Not super happy with the way I'm doing the backpropagation, would've loved to see it in a similar vein to how our forward propagation was handled zipping the layers with the activation function keys, but it works the way it is and the performance is almost identical. As stated in the patch notes, optimization is out of the window for this edition. I had some working prototypes but it greatly distracted from the readability and understandability. I should probably open an experimental branch, but I'll be opening a new issue that will cover these things in detail with future goals of the WBNN project.

**Comment by zMyst22** [comment]
Calling this one Case: CLOSED
Not super happy with the way I'm doing the backpropagation, would've loved to see it in a similar vein to how our forward propagation was handled zipping the layers with the activation function keys, but it works the way it is and the performance is almost identical. As stated in the patch notes, optimization is out of the window for this edition. I had some working prototypes but it greatly distracted from the readability and understandability. I should probably open an experimental branch, but I'll be opening a new issue that will cover these things in detail with future goals of the WBNN project.
